### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by Layrovell**
[DEMO LINK](https://Layrovell.github.io/react_static-list-of-todos/)

### CODE REVIEW

**Review by dtsehelnyk — APPROVED** [review]
Выглядит круто)
Даже адаптив есть 😮

File: src/App.js [hunk 1]
```diff
@@ -3,20 +3,28 @@ import './App.scss';
 
 import todos from './api/todos';
 import users from './api/users';
+import { TodoList } from './component/TodoList/TodoList';
+
+const prepareTodos = todos.map(todo => ({
```
Comment (aliceblunt8) [inline src/App.js:8]:
```suggestion
const preparedTodos = todos.map(todo => ({
```

File: src/component/Todo/Todo.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { User } from '../User/User';
+import './Todo.scss';
+
+export const Todo = ({ title, completed, user }) => {
+  const toUpperWord = title.replace(
+    /((?:^|[.?!])+\s*)(.)/g,
+    (m, tail, ch) => tail + ch.toUpperCase(),
+  );
```
Comment (aliceblunt8) [inline src/component/Todo/Todo.jsx:10]: create a function and move it outside the component

File: src/component/TodoList/TodoList.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export const TodoList = ({ todos }) => (
+  <ul className="todo__list">
+    {todos.map(item => (
+      <li className="todo__item" key={item.id}>
+        <Todo
+          title={item.title}
+          user={item.user}
+          completed={item.completed}
+        />
```
Comment (aliceblunt8) [inline src/component/TodoList/TodoList.jsx:14]: your version is also ok, but it's better to use spread operator
https://kevinyckim33.medium.com/jsx-spread-operator-component-props-meaning-3c9bcadd2493

```suggestion
        <Todo {...item}/>
```

File: src/component/TodoList/TodoList.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      </li>
+    ))}
+  </ul>
+);
+
+TodoList.defaultProps = {
+  todos: 0,
```
Comment (aliceblunt8) [inline src/component/TodoList/TodoList.jsx:21]: 0? why not an empty array?

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/component/Todo/Todo.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+          : <p className="status__todo not">not-completed</p>}
+      </div>
+      <div className="todo__name">
+        <User user={user} />
+      </div>
+    </div>
+  );
+};
+
+Todo.propTypes = TodoType;
```
Comment (aliceblunt8) [inline src/component/Todo/Todo.jsx:25]: describe here all todo props instead just adding TodoType, like this
```
Todo.propTypes = {
  userId: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  completed: PropTypes.bool.isRequired,
  user: UserType.isRequired,
}
```

File: src/component/Types/type.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    bs: PropTypes.string.isRequired,
+  }),
+});
+
+export const TodoType = {
+  userId: PropTypes.number.isRequired,
+  id: PropTypes.number.isRequired,
+  title: PropTypes.string.isRequired,
+  completed: PropTypes.bool.isRequired,
+  user: PropTypes.arrayOf(userTypes).isRequired,
```
Comment (aliceblunt8) [inline src/component/Types/type.jsx:32]: user is not an array, it's an object

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**